### PR TITLE
Test aws-secrets-manager extension with Localstack #3741

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/3741

This is not a complete coverage of the aws-secrets-manager component:

- Test for operation `replicateSecretToRegions` is commented, because it causes internal error on localStack (it might be caused by missing configuration - needs to be investigated)
- class `SecretsManagerPropertiesFunction` is not covered. As you can see [here](https://github.com/apache/camel/blob/main/components/camel-aws/camel-aws-secrets-manager/src/main/java/org/apache/camel/component/aws/secretsmanager/SecretsManagerPropertiesFunction.java#L91), aws client is configured from properties and url_override is not there. It has to be investigated how to cover this functionality.

I changed `quarkus/test/support/aws2/Aws2TestEnvContext.java]` to support secrets-manager, but it means that it supports **aws-** (not aws**2**-..) component. I think that better approach would be to rename module to support/aws/ or create a new module for aws(not 2). On the other hand it would be a bigger change because a few lines of the code.

I'm keeping this PR as draft and would like to ask, whether I should: 

1. Create separated module for support-aws(not 2)
2. Rename current suuport/aws2/ to support/aws/
3. Keep changes as it is

Before I mark this draft as prepared to merge, I'll create tickets for missing parts of the test coverage.

### CODE REVIEW

File: integration-tests-jvm/aws-secrets-manager/src/test/java/org/apache/camel/quarkus/component/aws/secrets/manager/it/AwsSecretsManagerTestEnvCustomizer.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public Service[] localstackServices() {
+        return new Service[] { Service.SECRETSMANAGER };
+    }
+
+    @Override
+    public void customize(Aws2TestEnvContext envContext) {
+
+        //get all properties from the context
+        Map<String, String> props = envContext.getProperies();
```
Comment (JiriOndrusek) [inline integration-tests-jvm/aws-secrets-manager/src/test/java/org/apache/camel/quarkus/component/aws/secrets/manager/it/AwsSecretsManagerTestEnvCustomizer.java:37]: @jamesnetherton I put the logic into the test customizer. Is this the place you are suggesting?

**Comment by JiriOndrusek** [comment]
The follow-up ticket is https://github.com/apache/camel-quarkus/issues/3848

File: integration-tests-jvm/aws-secrets-manager/src/test/java/org/apache/camel/quarkus/component/aws/secrets/manager/it/AwsSecretsManagerTestEnvCustomizer.java (hunk 1 above)
Comment (jamesnetherton) [reply integration-tests-jvm/aws-secrets-manager/src/test/java/org/apache/camel/quarkus/component/aws/secrets/manager/it/AwsSecretsManagerTestEnvCustomizer.java:37]: Yes, exactly.

Comment (JiriOndrusek) [reply integration-tests-jvm/aws-secrets-manager/src/test/java/org/apache/camel/quarkus/component/aws/secrets/manager/it/AwsSecretsManagerTestEnvCustomizer.java:37]: Thanks

**Comment by zbendhiba** [comment]
Hello @JiriOndrusek. What are the pending tasks for this PR?
Are there any responses for these questions mentioned in your PR message
```
1. Create separated module for support-aws(not 2)
2. Rename current suuport/aws2/ to support/aws/
3. Keep changes as it is
```

**Comment by JiriOndrusek** [comment]
@zbendhiba  This PR is prepared to be merged.
James suggested to move part of the code to different location, which renders my question obsolete.
I already switched from draft to ready for merge and i created a ticket with follow-up work - https://github.com/apache/camel-quarkus/issues/3848

**Comment by JiriOndrusek** [comment]
There might be problem with the failing CDI tests, I will look at it in the next days. (the change of the common part of aws-test-support seems like it shouldn't cause this error, but I'll verify it)

**Comment by zbendhiba** [comment]
> @zbendhiba This PR is prepared to be merged. James suggested to move part of the code to a different location, which renders my question obsolete. I already switched from draft to ready for merge and i created a ticket with follow-up work - #3848

thanks

**Review by oscerd — APPROVED** [review]

**Pull request merged**